### PR TITLE
adding resource delete tests

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Fixing behavior when trying to delete unexisting resource using **azure_rm_resource**

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
azure_rm_resource

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_resource.py [hunk 1]
```diff
@@ -286,7 +286,7 @@ def exec_module(self, **kwargs):
         needs_update = True
         response = None
 
-        if self.idempotency:
+        if self.idempotency or self.method == "DELETE":
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_resource.py:289]: self.idempotency always FALSE. what's the point of it?

File: test/integration/targets/azure_rm_resource/tasks/main.yml [hunk 2]
```diff
@@ -136,12 +136,39 @@
...
+    provider: Storage
+    resource_type: storageAccounts
+    resource_name: justunexisitingname
+    state: absent
+  register: output
+
+- debug:
+    var: output
+
+- name: Assert that no change was detected
```
Comment (yungezz) [inline test/integration/targets/azure_rm_resource/tasks/main.yml:153]: shouldn't the delete call fail?

File: lib/ansible/modules/cloud/azure/azure_rm_resource.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_resource.py:289]: idempotency option is just to enable idempotency check in present state, so i think it shouldn't be necessary to use this option when deleting resource.

File: test/integration/targets/azure_rm_resource/tasks/main.yml (hunk 2 above)
Comment (zikalino) [reply test/integration/targets/azure_rm_resource/tasks/main.yml:153]: if resource doesn't exist already and state is set to absent, it shouldn't fail.
this change is to make it consistent with ordinary modules.

**Comment by Fred-sun** [comment]
@yungezz @yuwzho Please help review this when you're available! Thanks!

**Comment by Fred-sun** [comment]
@yuwzho @yungezz  Please help review this PR and push to merged when your available! Thanks!

**Comment by Fred-sun** [comment]
@yuwzho @yungezz Please help review this PR when you're available! Thanks a lot!

**Comment by Fred-sun** [comment]
@zikalino How to move the PR going ahead? merged or waiting community member to review? Thank a lot!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@yungezz @zikalino Can we merged this PR or waiting others review? Thanks a lot!

**Comment by Fred-sun** [comment]
@MyronFanQiu Please help to review this PR and push forward to merge! It just add deleted test cases. Thank you very much!

**Comment by Fred-sun** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@yungezz @MyronFanQiu Please take a review when you're free! Thank you very much!

**Pull request closed without merging**